Mage.Sets/src/mage/cards/b/BrineHag.java

```diff
@@ -77,22 +77,10 @@ public boolean apply(Game game, Ability source) {
             return false;
         }
 
-        List<Permanent> list = new ArrayList<>();
-        for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
-            Player player = game.getPlayer(playerId);
-            if (player == null) {
-                continue;
-            }
-
-            for (Permanent creature : game.getBattlefield().getAllActivePermanents(StaticFilters.FILTER_PERMANENT_CREATURE, playerId, game)) {
-                if (sourcePermanent.getDealtDamageByThisTurn().contains(new MageObjectReference(creature.getId(), game))) {
-                    list.add(creature);
-                }
-            }
-        }
-        if (!list.isEmpty()) {
+        Set<MageObjectReference> set = new HashSet<>(sourcePermanent.getDealtDamageByThisTurn());
```

Comment:
is there a reason for the `new HashSet<>` here? doesn't `getDealtDamageByThisTurn()` return a set already?

Reply:
I was worried that the set could be modified by something else. I guess it can't in this case because the permanent's already dead, but I prefer the defensive programming like this, especially since I haven't paid close attention to exactly how the rollback code works.

Reply:
Makes sense